types/react-instantsearch-core/index.d.ts

```diff
@@ -193,9 +185,12 @@ export interface AutocompleteExposed {
 }
 
 // tslint:disable-next-line:no-unnecessary-generics
-export function connectAutoComplete<TDoc = BasicDoc>(stateless: React.StatelessComponent<AutocompleteProvided<TDoc>>): React.ComponentClass<AutocompleteExposed>;
-export function connectAutoComplete<Props extends AutocompleteProvided<TDoc>, TDoc = BasicDoc>(Composed: React.ComponentType<Props>):
-  ConnectedComponentClass<Props, AutocompleteProvided<TDoc>, AutocompleteExposed>;
+export function connectAutoComplete<TDoc = BasicDoc>(
+  stateless: React.StatelessComponent<AutocompleteProvided<TDoc>>
```

Comment:
I think the linter is complaining because the generic `<TDoc>` is useless here.  I can do `connectAutoComplete(MyComponent)` regardless of what type of hit my component uses, and it doesn't affect the output.  This might as well be `React.StatelessComponent<AutocompleteProvided<any>>`.

Now, in this case I disagree with the linter because for stateless components the type inference can be extremely helpful in the "BasicDoc" case.  Consider this example:
```tsx
connectAutoComplete(({ hits }) => {
   
  return <ul>
      // hit._highlightResult.title is a HighlightResultPrimitive
     {hits.map((hit) => <li dangerouslySetInnerHtml={{ __html: hit._highlightResult.title.value }}></li>}
  </ul>
})
```

The inferred `TDoc` is `BasicDoc` which not only allows this construction, it provides excellent intellisense support for it.  If the developer has a more complex document they can specify it in the type of the anonymous function's first argument.

Anyway, that is my reasoning for why on line 187 I had put `// tslint:disable-next-line:no-unnecessary-generics`.  But now that you've split it over two lines it's triggering `no-unnecesary-generics` again!